architecture/adr-007-use-c4.md

```diff
@@ -10,14 +10,21 @@ Although primarily aimed at software architects and developers, the C4 model pro
 
 The C4 model is a lean graphical notation technique for modelling the architecture of software systems, an abstraction first approach to diagramming architecture, based upon abstractions that reflect how software architects and developers think about and build software. The deliberately small set of abstractions and diagram types is what makes the C4 model easy to learn and use.
 
-Good software architecture diagrams assist with communication (both inside and outside the development and product teams), onboarding new staff, easy of contribution, risk identification, threat modelling, etc.
+Good software architecture diagrams assist with communication (both inside and outside the development and product teams), onboarding new staff, easy of contribution, risk identification, threat modelling, etc. Also, good software architecture diagrams help to align everybody's understanding of the software being built, helping therefore the team to be more efficient.
+
+When it comes to actually to render C4, once you have the architecture models composed of the common SDL format, there are a number of tools available to render them, like PlantUML, Mermaid, Structurizr, and more.
 
 ## Decision
 
 Within ComposeUI, we are to position C4 as the preferred way to capture system architecture and how it is deployed, both as a a means to creating consistent and easily to understand diagrams as code that can live in the source control, and also as a data model that can be used to feed static analysis systems thereby reducing the overhead for developers and improving accuracy of the decisions and suggestions of those systems.
 
+To describe the actual C4, we would use PlantUML extensions and rendering.

```

Comment:
Why PlantUML?

Reply:
Sure, what else? You prefer Mermaid?

Reply:
Scratch that, now that I installed it for eval, I can see that no server is needed.

Reply:
Indeed, wanted to explain in details that you do not need a server - you can just store the DSL in your source control and when docs are built to http://opensource.morganstanley.com/ComposeUI/, that would convert them (and also VSCode plugins do that as you are editing; we can have a compromise solution as well that we do check in the generated png along with the DSL so the pictures would be visible also when you are browsing the github site).